airgun/views/host_new.py

```diff
@@ -264,12 +266,84 @@ class traces(Tab):
         enable_traces = OUIAButton('OUIA-Generated-Button-primary-1')
 
     @View.nested
-    class ansible(Tab):
+    class insights(Tab):
         pass
 
     @View.nested
-    class insights(Tab):
-        pass
+    class ansible(SatTab):
+        """View comprising the subtabs under the Ansible Tab"""
+        ROOT = './/div[contains(@class, "ansible-host-details")]'
+        TAB_NAME = 'Ansible'
+
+        @View.nested
+        class roles(SatTab):
+            TAB_NAME = 'Roles'
+            assignedRoles = Text('.//a[contains(@href, "roles/all")]')
+            edit = Button(locator='.//button[@aria-label="edit ansible roles"]')
+            table = PatternflyTable(
+                #No OUIA ID
+                locator='',
+                column_widgets={
+                    'Name': Text('.//a')
+                },
+            )
+
+        @View.nested
+        class variables(SatTab):
+            TAB_NAME = 'Variables'
+            table = PatternflyTable(
+                #No OUIA ID
+                locator='',
+                column_widgets={
+                    'Name': Text('.//a'),
+                    'Ansible role': Text('./span'),
+                    'Type': Text('./span'),
+                    #the next field can also be a form group
+                    'Value': Text('./span'),
+                    'Source attribute': Text('./span'),
+                    #The next 2 buttons are hidden by default, but appear in this order
+                    5: Button(locator='.//button[@aria-label="Cancel editing override button"]'),
+                    6: Button(locator='.//button[@aria-label="Submit override button"]'),
+                    #Clicking this button hides it, and displays the previous 2
+                    7: Button(locator='.//button[@aria-label="Edit override button"]')
+                },
+            )
+
+        @View.nested
+        class inventory(SatTab):
+            TAB_NAME = 'Inventory'
+            ROOT = ""
+
+        @View.nested
+        class jobs(SatTab):
```

Comment:
In chat, @mayurilahane pointed out that representing these three states in a single view class could cause issues with `is_displayed`. Are you planning to keep this class as is or split it into separate classes?

Reply:
also, you will need to add "page refresh" as a prerequisite to the second view 
ex: https://gitlab.cee.redhat.com/insights-qe/iqe-ros-plugin/-/blob/master/iqe_ros/entities/__init__.py#L147

Reply:
also, consider page refresh time